File: second-year/second-semester/MPP/mpp-proiect-repository-raulbaciulescu/travelAgency2/travelAgency2/Repository/Repository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using travelAgency2.src.Domain;

namespace travelAgency2.src.Repository
{
    internal interface Repository<ID, T> where T : Entity<ID>
    {
        void Add(T entity);
        void Update(T entity, T newEntity);
        T FindByID(ID id);
        void Delete(ID id);
        List<T> GetAll();
    }
}

File: second-year/second-semester/MPP/examen/rezolvari/examen_curs/Persistence/src/main/java/api/Repository.java
package api;

import domain.Entity;

import java.util.List;
import java.util.Optional;

public interface Repository<ID, T extends Entity<ID>> {
    void add(T entity);
    void update (T entity, T newEntity);
    void delete(ID id);
    Optional<T> findByID(ID id);
    List<T> getAll();
}
